### PR TITLE
feat: modify edit generate api token page

### PR DESCRIPTION
**Pull request by rish2320**
# Description

Modify the edit and generate token page and convert the card format to a page.

Fixes # [2477](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=2477)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
1: To start with the page has been tested by manually checking the UI and functionalities on the front-end.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/apiTokens/CreateAPIToken.tsx [hunk 1]
```diff
@@ -207,21 +211,45 @@ function CreateAPIToken({
...
     }
 
+    const handleQuestion = () => {
+        return (
+            <TippyCustomized
+                theme={TippyTheme.white}
+                className="w-300 h-100 fcv-5"
+                placement="right"
+                Icon={QuestionFilled}
+                heading={"API tokens"}
```
Comment (milindk8) [inline src/components/apiTokens/CreateAPIToken.tsx:221]: Create const for all the messaging

File: src/components/apiTokens/CreateAPIToken.tsx [hunk 2]
```diff
@@ -207,21 +211,45 @@ function CreateAPIToken({
...
+                infoText="API tokens are like ordinary OAuth access tokens. They can be used instead of username and password for programmatic access to API."
```
Comment (milindk8) [inline src/components/apiTokens/CreateAPIToken.tsx:222]: same here

File: src/components/apiTokens/CreateAPIToken.tsx [hunk 3]
```diff
@@ -207,21 +211,45 @@ function CreateAPIToken({
...
-                    API tokens
-                </span>{' '}
-                / New API token
+        <div className='bcn-0 api__token'>
+            <div className='flex dc__content-space pt-16 pb-16 dc__gap-8'>
+                <div className='flex row ml-0 h-32'>
+                    <div className="cn-9 fw-6 fs-16">
+                        <span className="cb-5 cursor" onClick={redirectToTokenList}>
+                            API tokens
+                        </span>{' '}
```
Comment (milindk8) [inline src/components/apiTokens/CreateAPIToken.tsx:244]: remove this

File: src/components/apiTokens/CreateAPIToken.tsx [hunk 4]
```diff
@@ -207,21 +211,45 @@ function CreateAPIToken({
...
-                <span className="cb-5 cursor" onClick={redirectToTokenList}>
-                    API tokens
-                </span>{' '}
-                / New API token
+        <div className='bcn-0 api__token'>
+            <div className='flex dc__content-space pt-16 pb-16 dc__gap-8'>
+                <div className='flex row ml-0 h-32'>
+                    <div className="cn-9 fw-6 fs-16">
+                        <span className="cb-5 cursor" onClick={redirectToTokenList}>
+                            API tokens
```
Comment (milindk8) [inline src/components/apiTokens/CreateAPIToken.tsx:243]: create const

File: src/components/apiTokens/CreateAPIToken.tsx [hunk 5]
```diff
@@ -207,21 +211,45 @@ function CreateAPIToken({
...
+                        </span>{' '}
+                        / New API token
```
Comment (milindk8) [inline src/components/apiTokens/CreateAPIToken.tsx:245]: Same for this

File: src/components/apiTokens/EditAPIToken.tsx [hunk 6]
```diff
@@ -226,22 +231,58 @@ function EditAPIToken({
...
     }
 
+    const handleQuestion = () => {
+        return (
+            <TippyCustomized
+                theme={TippyTheme.white}
+                className="w-300 h-100 fcv-5"
+                placement="right"
+                Icon={QuestionFilled}
+                heading={"API tokens"}
```
Comment (milindk8) [inline src/components/apiTokens/EditAPIToken.tsx:241]: Use common const here

File: src/components/apiTokens/EditAPIToken.tsx [hunk 7]
```diff
@@ -226,22 +231,58 @@ function EditAPIToken({
...
-                <span className="cb-5 cursor" onClick={redirectToTokenList}>
-                    API tokens
-                </span>{' '}
-                / Edit API token
+        <div className="fs-13 fw-4 api__token">
+            <div className='flex dc__content-space pt-16 pb-16 dc__gap-8'>
+                <div className='flex row ml-0'>
+                    <div className="cn-9 fw-6 fs-16">
+                        <span className="cb-5 cursor" onClick={redirectToTokenList}>
+                            API tokens
```
Comment (milindk8) [inline src/components/apiTokens/EditAPIToken.tsx:263]: Use same const here from previous file

File: src/components/apiTokens/EditAPIToken.tsx [hunk 8]
```diff
@@ -226,22 +231,58 @@ function EditAPIToken({
...
+                        </span>{' '}
+                        / Edit API token
```
Comment (milindk8) [inline src/components/apiTokens/EditAPIToken.tsx:265]: here too

File: src/css/base.scss [hunk 9]
```diff
@@ -2453,6 +2488,14 @@ textarea,
 
 // Padding in pixels
 
+.p-0-imp {
+    padding: 0px !important;
+}
+
+.pl-0-imp {
```
Comment (milindk8) [inline src/css/base.scss:2491]: Please remove all of this, It's already available in formula.scss

File: src/components/apiTokens/CreateAPIToken.tsx (hunk 1 above)
Comment (rish2320) [reply src/components/apiTokens/CreateAPIToken.tsx]: done

File: src/components/apiTokens/CreateAPIToken.tsx (hunk 2 above)
Comment (rish2320) [reply src/components/apiTokens/CreateAPIToken.tsx]: done

File: src/components/apiTokens/CreateAPIToken.tsx (hunk 3 above)
Comment (rish2320) [reply src/components/apiTokens/CreateAPIToken.tsx:245]: done

File: src/components/apiTokens/CreateAPIToken.tsx (hunk 4 above)
Comment (rish2320) [reply src/components/apiTokens/CreateAPIToken.tsx]: done

File: src/components/apiTokens/CreateAPIToken.tsx (hunk 5 above)
Comment (rish2320) [reply src/components/apiTokens/CreateAPIToken.tsx]: done

File: src/components/apiTokens/EditAPIToken.tsx (hunk 6 above)
Comment (rish2320) [reply src/components/apiTokens/EditAPIToken.tsx]: done

File: src/components/apiTokens/EditAPIToken.tsx (hunk 7 above)
Comment (rish2320) [reply src/components/apiTokens/EditAPIToken.tsx]: done

File: src/components/apiTokens/EditAPIToken.tsx (hunk 8 above)
Comment (rish2320) [reply src/components/apiTokens/EditAPIToken.tsx]: done

File: src/css/base.scss (hunk 9 above)
Comment (rish2320) [reply src/css/base.scss]: done

File: src/components/apiTokens/GenerateActionButton.tsx [hunk 10]
```diff
@@ -34,7 +21,7 @@ function GenerateActionButton({
                     Cancel
                 </ButtonWithLoader>
                 <ButtonWithLoader
-                    rootClassName="flex cta h-36"
+                    rootClassName= {`flex cta h-36 ${regenerateButton ? 'second' : 'mr-12 first'}`}
                     onClick={() => onSave()}
```
Comment (milindk8) [inline src/components/apiTokens/GenerateActionButton.tsx:25]: {onSave}

File: src/config/constantMessaging.ts [hunk 11]
```diff
@@ -164,4 +164,13 @@ export const APP_GROUP_CI_DETAILS = {
 
 export const DEPLOYMENT_HISTORY_TABS = {
   SOURCE : 0,
+}
+
+export const API_COMPONENTS = {
+
+    title: "API tokens",
```
Comment (milindk8) [inline src/config/constantMessaging.ts:171]: Write key in uppercase

File: src/components/apiTokens/GenerateActionButton.tsx (hunk 10 above)
Comment (rish2320) [reply src/components/apiTokens/GenerateActionButton.tsx]: done

File: src/config/constantMessaging.ts (hunk 11 above)
Comment (rish2320) [reply src/config/constantMessaging.ts]: done

File: src/components/apiTokens/CreateAPIToken.tsx [hunk 12]
```diff
@@ -207,21 +212,45 @@ function CreateAPIToken({
         setAdminPermission(e.target.value)
     }
 
+    const handleQuestion = () => {
```
Comment (milindk8) [inline src/components/apiTokens/CreateAPIToken.tsx:215]: If its possible create common function somewhere else & use on both the place

Comment (rish2320) [reply src/components/apiTokens/CreateAPIToken.tsx]: done

**Review by milindk8 — APPROVED** [review]

File: src/css/formulae.scss [hunk 13]
```diff
@@ -187,6 +187,7 @@ $color-suffix: 00;
 }
 
 .cta,
+
```
Comment (vivek-devtron) [inline src/css/formulae.scss:190]: Please remove this Space

File: src/css/base.scss [hunk 14]
```diff
@@ -1725,6 +1756,10 @@ button.anchor {
     border-top: solid 1px var(--N200);
 }
 
+.dc__border-top-1 {
```
Comment (vivek-devtron) [inline src/css/base.scss:1759]: we can use .dc__border-top-n1 instead creating new 1

File: src/components/apiTokens/RegenerateModal.tsx [hunk 15]
```diff
@@ -133,6 +133,7 @@ function RegeneratedModal({
                     onCancel={() => setShowRegeneratedModal(false)}
                     onSave={handleRegenrateToken}
                     buttonText="Regenerate Token"
+                    regenerateButton = {true}
```
Comment (vivek-devtron) [inline src/components/apiTokens/RegenerateModal.tsx:136]: format this line

File: src/components/apiTokens/CreateAPIToken.tsx [hunk 16]
```diff
@@ -208,20 +236,21 @@ function CreateAPIToken({
...
-                    API tokens
-                </span>{' '}
-                / New API token
+        <div className='bcn-0 api__token'>
+            <div className='flex dc__content-space pt-16 pb-16 dc__gap-8'>
+                <div className='flex row ml-0 h-32'>
+                    <div className="cn-9 fw-6 fs-16">
+                        <span className="cb-5 cursor" onClick={redirectToTokenList}>
+                            {API_COMPONENTS.TITLE}
+                        </span>{' '}
```
Comment (vivek-devtron) [inline src/components/apiTokens/CreateAPIToken.tsx:245]: Please remove {' '}

File: src/config/constantMessaging.ts [hunk 17]
```diff
@@ -169,4 +169,13 @@ export const DEPLOYMENT_HISTORY_TABS = {
 export const CONFIGMAPS_SECRETS = {
     configmaps: "configmaps",
     secrets: "secrets"
+}
+
+export const API_COMPONENTS = {
+
```
Comment (shivani170) [inline src/config/constantMessaging.ts:175]: Remove extra lines from every where

File: src/components/apiTokens/CreateAPIToken.tsx [hunk 18]
```diff
@@ -28,6 +30,32 @@ import ExpirationDate from './ExpirationDate'
...
+            Icon={QuestionFilled}
+            heading={API_COMPONENTS.TITLE}
+            infoText={API_COMPONENTS.QUESTION_ICON_INFO}
+            showCloseButton={true}
+            trigger="click"
+            interactive = {true}
+            documentationLink={DOCUMENTATION.WEBHOOK_API_TOKEN}
+            documentationLinkText="View Documentation"
+        >
+            <div className="icon-dim-20 fcn-9 ml-8 cursor">
```
Comment (shivani170) [inline src/components/apiTokens/CreateAPIToken.tsx:52]: Can directly apply these classname in <Question/>

File: src/components/apiTokens/CreateAPIToken.tsx [hunk 19]
```diff
@@ -208,20 +236,21 @@ function CreateAPIToken({
...
+        <div className='bcn-0 api__token'>
+            <div className='flex dc__content-space pt-16 pb-16 dc__gap-8'>
+                <div className='flex row ml-0 h-32'>
+                    <div className="cn-9 fw-6 fs-16">
+                        <span className="cb-5 cursor" onClick={redirectToTokenList}>
+                            {API_COMPONENTS.TITLE}
+                        </span>{' '}
+                        {API_COMPONENTS.NEW_API_TITLE}
+                    </div>
+                    {handleQuestion()}
```
Comment (shivani170) [inline src/components/apiTokens/CreateAPIToken.tsx:248]: Please rename it to renderQuestionwithTippy or something better

File: src/css/base.scss [hunk 20]
```diff
@@ -856,6 +856,37 @@ button.anchor {
     background-color: #f5f5f5;
 }
 
+.override-button.cta {
```
Comment (shivani170) [inline src/css/base.scss:859]: Is it used anywhere in the dashboard? if not can remove

File: src/components/apiTokens/GenerateActionButton.tsx [hunk 21]
```diff
@@ -34,16 +21,15 @@ function GenerateActionButton({
                     Cancel
                 </ButtonWithLoader>
                 <ButtonWithLoader
-                    rootClassName="flex cta h-36"
-                    onClick={() => onSave()}
+                    rootClassName= {`flex cta h-36 ${regenerateButton ? 'second' : 'mr-12 first'}`}
```
Comment (shivani170) [inline src/components/apiTokens/GenerateActionButton.tsx:24]: can we rename first &b second like `order-first`  or more specific

File: src/components/apiTokens/RegenerateModal.tsx (hunk 15 above)
Comment (rish2320) [reply src/components/apiTokens/RegenerateModal.tsx:136]: done

File: src/components/apiTokens/EditAPIToken.tsx [hunk 22]
```diff
@@ -227,21 +229,34 @@ function EditAPIToken({
...
-                    API tokens
-                </span>{' '}
-                / Edit API token
+        <div className="fs-13 fw-4 api__token">
+            <div className='flex dc__content-space pt-16 pb-16 dc__gap-8'>
+                <div className='flex row ml-0'>
+                    <div className="cn-9 fw-6 fs-16">
+                        <span className="cb-5 cursor" onClick={redirectToTokenList}>
+                            {API_COMPONENTS.TITLE}
+                        </span>{' '}
```
[the remark this answers is not in the document]
Comment (rish2320) [reply src/components/apiTokens/EditAPIToken.tsx:238]: done

File: src/components/apiTokens/CreateAPIToken.tsx (hunk 16 above)
Comment (rish2320) [reply src/components/apiTokens/CreateAPIToken.tsx:245]: done

File: src/config/constantMessaging.ts (hunk 17 above)
Comment (rish2320) [reply src/config/constantMessaging.ts:175]: done

File: src/components/apiTokens/GenerateActionButton.tsx (hunk 21 above)
Comment (rish2320) [reply src/components/apiTokens/GenerateActionButton.tsx:24]: done

File: src/css/base.scss (hunk 20 above)
Comment (rish2320) [reply src/css/base.scss:859]: Yes it is used.

File: src/components/apiTokens/CreateAPIToken.tsx (hunk 19 above)
Comment (rish2320) [reply src/components/apiTokens/CreateAPIToken.tsx:248]: done

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**